fix(vaadin-grid): Material Grid Improvements

platosha opened the pull request:
Fixes vaadin/vaadin-grid#807

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/vaadin/vaadin-material-theme/45)
<!-- Reviewable:end -->

jouni commented:
<img width="827" alt="screen shot 2018-05-21 at 9 25 45" src="https://user-images.githubusercontent.com/66382/40293301-fa28631e-5cd8-11e8-9e22-e6b3d89f7c00.png">

When you resize the columns as small as possible, the contents overflow the cells.

jouni commented on vaadin-grid-sorter.html:
```diff
@@ -0,0 +1,71 @@
+<link rel="import" href="color.html">
+<link rel="import" href="font-icons.html">
+<link rel="import" href="typography.html">
```
Unnecessary import.

platosha replied:
Good catch, removed.

jouni commented on test/all-theme-imports.html:
```diff
@@ -10,6 +10,12 @@
 <link rel="import" href="../vaadin-date-picker.html">
 <link rel="import" href="../vaadin-dialog.html">
 <link rel="import" href="../vaadin-dropdown-menu.html">
+<link rel="import" href="../vaadin-grid-column.html">
```
No import for `vaadin-grid.html`?

platosha replied:
It’s there below. On my system, `ls` does sort the files this way. I can’t notice any errors with that order, so I’d keep it as it is.

jouni commented:
<img width="985" alt="screen shot 2018-05-21 at 9 34 09" src="https://user-images.githubusercontent.com/66382/40293762-128452fe-5cdb-11e8-97ef-e43cbf7b39d0.png">

Is the selection color applied for the detail row on purpose? See https://github.com/vaadin/vaadin-grid/issues/1197 for reference.

platosha commented:
>  Do we want to introduce the dense variant already? I removed it from the button, and was planning on doing the same for text field. I think it might make sense to have a more solid concept around the density, similar to what we have in Lumo with sizing & spacing.

Yeah maybe not. Removed that. It’s a very neat feature for the grid, however.

> When you resize the columns as small as possible, the contents overflow the cells.

Fixed.

> Is the selection color applied for the detail row on purpose? See vaadin/vaadin-grid#1197 for reference.

Actually that was accidental. Removed for now, so that the “hover” and “selection” states are applied the same way, only to the data row.

I have considered to keep applying the selection highlight to the detail row too, but decided against that idea. Because the mouse click on the detail row does not actually change selection in the demo, applying selection highlight on the detail row is a bit confusing.

jouni commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 11 of 15 files at r1, 4 of 4 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io/reviews/vaadin/vaadin-material-theme/45#-:-LDFonUu5lGIgsIBamn3:bnfp4nl)*
<!-- Sent from Reviewable.io -->
